src/login/msal/MsalAuthProvider.ts

```diff
@@ -87,9 +81,8 @@ export class MsalAuthProvider extends AuthProviderBase<AuthenticationResult> {
 		}
 	}
 
-	public getCredentials(environment: string, userId: string, tenantId: string): DeviceTokenCredentials {
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
-		return new DeviceTokenCredentials({ environment: (<any>Environment)[environment], username: userId, clientId, tokenCache: this.dummyTokenCache, domain: tenantId });
+	public getCredentials(): undefined {
+		return undefined;
```

Comment:
I changed this back to `return undefined` because throwing an error would prevent the cached MSAL session from being loaded in `initializeSessions`. Do you know if there's a way to only throw an error only if that credentials object is accessed?

Reply:
You'd want to use a getter. You can do than on any object, but the cleanest option would likely be to create an `AzureSession` class